### ISSUE TITLE
ansible-lint tries to parse README.md and fails

### ISSUE THREAD

**Issue by alexbarton** [ansible-community/ansible-lint#1487]
##### Summary

I have a directory structure like this:

```
ansible
├── plays
│   ├── README.md
│   ├── deploy
│   ├── site
│   […]
├── roles
│   ├── README.md
│   […]
[…]
```

And `ansible-lint` tries to parse the `README.md` Markdown files – and fails:

```bash
ansible-lint --show-relpath -v
INFO     Added ANSIBLE_ROLES_PATH=~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:roles
INFO     Discovering files to lint: git ls-files -z
INFO     Discovering files to lint: git ls-files -z
Failed to parse YAML in roles/README.md: did not find expected <document start>
  in "<unicode string>", line 9, column 1
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
❯ ansible --version
ansible 2.9.19
  config file = None
  configured module search path = ['/Users/…/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/…/.venv/lib/python3.9/site-packages/ansible
  executable location = /Users/…/.venv/bin/ansible
  python version = 3.9.2 (default, Mar 15 2021, 17:27:56) [Clang 12.0.0 (clang-1200.0.32.29)]

❯ ansible-lint --version
ansible-lint 5.0.4 using ansible 2.9.19
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

macOS 10.15.7

##### Desired Behaviour

I think `ansible-lint` should skip Markdown files `*.md` altogether?

**Comment by ssbarnea** [comment]
Historically it did look only at yaml/yml files but we are going into a direction where it could process other file types, like rst or md, as they can also need to follow some rules.

Still, it should never try to load an md file as YAML for sure. Please pin it down until I release a fix for it.

**Comment by alexbarton** [comment]
I can confirm that this is fixed. Thanks a lot @ssbarnea!
